libs/block-editor/src/lib/plugins/floating.plugin.ts

```diff
@@ -1,71 +1,106 @@
+    clientRect: (() => DOMRect) | null;
+}
 
-    public preventHide = false;
+export interface FloatingActionsKeydownProps {
+    view: EditorView;
+    event: KeyboardEvent;
+    range: Range;
+}
 
-    public tippy!: Instance;
+export interface FloatingRenderActions {
+    onStart?: (props: FloatingActionsProps) => void;
+    onExit?: (props: FloatingActionsProps) => void;
+    onKeyDown?: (props: FloatingActionsKeydownProps) => boolean;
+}
 
-    public command: (props: { rect: DOMRect, range: Range, editor: Editor }) => void;
+export const FLOATING_ACTIONS_MENU_KEYBOARD = 'menuFloating';
 
-    constructor({ editor, element, view, tippyOptions, on: { command } }: FloatingActionsViewProps) {
-        console.log('constructor');
+export class FloatingActionsView {
+    editor: Editor;
+    element: HTMLElement;
+    view: EditorView;
+    tippy!: Instance;
+    render: () => FloatingRenderActions;
+    command: (props: { editor: Editor; range: Range; props: SuggestionsCommandProps }) => void;
+    key: PluginKey;
+
+    constructor({
+        editor,
+        element,
+        view,
+        tippyOptions,
+        render,
+        command,
+        key
+    }: FloatingActionsViewProps) {
         this.editor = editor;
         this.element = element;
         this.view = view;
         this.element.addEventListener('mousedown', this.mousedownHandler, { capture: true });
-        this.editor.on('focus', this.focusHandler);
-        this.editor.on('blur', this.blurHandler);
-        this.createTooltip(tippyOptions);
         this.element.style.visibility = 'visible';
+        this.render = render;
         this.command = command;
+        this.key = key;
+        this.createTooltip(tippyOptions);
     }
 
-    mousedownHandler = () => {
-        console.log('mousedownHandler');
-        this.preventHide = true;
-
-        const { selection } = this.editor.state;
-        const { from, to } = selection;
-        const rect = posToDOMRect(this.view, from, to);
-        this.command({ rect, range: { from, to }, editor: this.editor });
-    };
-
-    focusHandler = () => {
-        console.log('focusHandler');
-        // we use `setTimeout` to make sure `selection` is already updated
-        setTimeout(() => this.update(this.editor.view));
-    };
-
-    blurHandler = () => {
-        console.log('blurHandler');
-        this.view.focus()
+    /**
+     * Element mousedown handler to update the plugin state and open
+     *
+     * @param {MouseEvent} e
+     * @memberof FloatingActionsView
+     */
+    mousedownHandler = (e: MouseEvent): void => {
+        e.preventDefault();
+
+        const transaction = this.editor.state.tr.setMeta(FLOATING_ACTIONS_MENU_KEYBOARD, {
+            open: true
+        });
+        this.editor.view.dispatch(transaction);
     };
 
-
-    createTooltip(options: Partial<Props> = {}) {
+    /**
+     * Create the tooltip for the element
+     *
+     * @param {Partial<Props>} [options={}]
```

Comment:
`options` not optional

Reply:
Not sure but this is how Document This generate it

Reply:
mmm weird
![image](https://user-images.githubusercontent.com/37185433/130681988-042b74a4-05f3-4db0-8a12-9f7e3cfadd9c.png)